CRM/Contact/Form/Search/Custom/PriceSet.php

```diff
@@ -316,6 +354,7 @@ public function from() {
     $from = "
 FROM       civicrm_contact contact_a
 INNER JOIN {$this->_tableName} tempTable ON ( tempTable.contact_id = contact_a.id ) {$this->_aclFrom}
+INNER JOIN civicrm_participant_status_type participant_status ON (tempTable.status_id = participant_status.id)
```

Comment:
Doing this as a join may hurt performance. In other searches we look up the options with PHP when rendering results.